src/pages/UserPages/UserWallet/index.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import Authenticate from '../../../components/Authenticate'
+import { getTransaction } from '../../../store/reducers/user'
+import PageNotFound from '../../PageNotFound'
+import Wallet from './Wallet'
+import Footer from '../../../components/Footer'
+import { Table } from '../../../components/Table'
+import { thousands_separators, timeAgo, TIME_ZONE } from '../../../utils/helperFunctions'
+import BaseModal from '../../../components/Modal/BaseModal'
+import { useState } from 'react'
+
+
+const UserWallet = () => {
+
+    const { role, isLoading, isApproved, wallet, uid, walletTransactions } = useSelector(state => ({
+        role : state.user.role,
+        isLoading : state.isLoading,
+        isApproved : state.user.isApproved,
+        wallet : state.user.wallet,
+        uid : state.user.uid,
+        walletTransactions : state.user.walletTransactions
+    }))
+
+    const [show, setShow] = useState(false)
+    const [modalContent, setModalContent] = useState({})
+
+    const dispatch = useDispatch()
+
+    const setContent = (title, size, content) => {
+        setModalContent({
+            title,
+            size,
+            content
+        })
+
+        setShow(true)
+    }
+
+
+    useEffect(() => {
+
+        dispatch(getTransaction(uid))
+
+    }, [uid, dispatch])
```

Comment:
we do not need to have `dispatch` here

Reply:
It really doesn't hurt... `useEffect` gives a warning on. if I didn't have to have `uid` in the dependency array, I would have added an eslint breaker there... but since I need the `uid` in the dependency array, I'm also adding `dispatch` to shut Firebase warning up... Doesn't affect the code in any way